src/com/google/common/css/MultipleMappingSubstitutionMap.java

```diff
@@ -76,5 +77,5 @@ public static ValueWithMappings createForSingleMapping(String key, String value)
    * associated with this substitution. These mappings are available as
    * {@link ValueWithMappings#mappings}.
    */
-  public ValueWithMappings getValueWithMappings(String key);
+  public ValueWithMappings getValueWithMappings(String key, @Nullable Map<String, String> inputMappings);
```

Comment:
I'm trying to figure out the intent for adding this parameter. It appears that you're trying to prevent mappings that appear in the input map from being present in the output map. Is that correct?

If so, I think it would it be easier to just post-filter the output map by removing the duplicates. This would even allow us to have this behavior be optional. For example, if you wanted stable renaming, the output map of the previous release would be the input map of the next release.

What do you think?